frontend/src/components/project/ProjectDeleteDialog.tsx

```diff
@@ -0,0 +1,172 @@
+          cancelledMessage: t('Cancelled deletion of project {{ projectName }}.', {
+            projectName: project.id,
+          }),
+          successMessage: t('Deleted project {{ projectName }}.', { projectName: project.id }),
+          errorMessage: t('Error deleting project {{ projectName }}.', { projectName: project.id }),
+          successUrl: '/',
+        }
+      )
+    );
+
+    onClose();
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onClose}
+      aria-labelledby="project-delete-dialog-title"
+      aria-describedby="project-delete-dialog-description"
+      maxWidth="sm"
+      fullWidth
+    >
+      <DialogTitle id="project-delete-dialog-title">{t('Delete Project')}</DialogTitle>
+      <DialogContent>
+        <DialogContentText id="project-delete-dialog-description" component="div">
+          <Typography variant="body1" paragraph>
+            {t('What would you like to do with project "{{projectName}}"?', {
+              projectName: project.id,
+            })}
+          </Typography>
+
+          <Typography variant="body2" paragraph>
+            {t(
+              'By default, this will only remove the project label from the following namespaces:'
+            )}
+          </Typography>
+
+          <ul>
+            {project.namespaces.map(namespaceName => (
+              <li key={namespaceName}>
+                <strong>{namespaceName}</strong>
+              </li>
+            ))}
+          </ul>
+
+          <AuthVisible item={namespaces[0]} authVerb="delete">
+            <FormControlLabel
+              control={
+                <Checkbox
+                  checked={deleteNamespaces}
+                  onChange={e => setDeleteNamespaces(e.target.checked)}
+                  name="deleteNamespaces"
+                  color="primary"
+                />
+              }
+              label={
+                <Typography variant="body2">
+                  {t('Also delete the namespaces (this will remove all resources within them)')}
+                </Typography>
+              }
+            />
+
+            {deleteNamespaces && (
+              <Typography variant="body2" color="error" sx={{ mt: 1, fontWeight: 'bold' }}>
+                {t(
+                  'Warning: This action cannot be undone. All resources in these namespaces will be permanently deleted.'
+                )}
+              </Typography>
+            )}
+          </AuthVisible>
+        </DialogContentText>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={onClose} color="secondary" variant="contained">
+          {t('Cancel')}
+        </Button>
+        <Button
+          onClick={handleDelete}
+          color="primary"
+          variant="contained"
+          sx={{ minWidth: '200px' }} // Fixed width to prevent button resize
+        >
+          {deleteNamespaces ? t('Delete Project & Namespaces') : t('Remove Project')}
```

Comment:
Maybe it makes more sense to stay consistent with the term "Delete", WDYT?
```suggestion
          {deleteNamespaces ? t('Delete Project & Namespaces') : t('Delete Project')}
```